### ISSUE TITLE
Strange symbols are added when commenting an already commented line

### ISSUE THREAD

**Issue by himat** [scrooloose/nerdcommenter#294]
Just installed the plugin, and tried it in a C file.

I first just used <leader>cc to comment the line seen below.
```
for(i=1; i <= n1 && i <= n2; ++i)
{
     /*Checks if i is factor of both integers*/
     if(n1%i==0 && n2%i==0)
          gcd = i;
}
```

Then, I selected the entire loop, and used <leader>cc again.
```
  /*for(i=1; i <= n1 && i <= n2; ++i)*/
  /*{*/
       /*[>Checks if i is factor of both integers<]*/
      /*if(n1%i==0 && n2%i==0)*/
          /*gcd = i;*/
  /*}*/
```

But now, as you can see, the prevent comment now has [> and <]. Why is this happening? Even if I then uncomment the entire block using <leader>c<space>, the new symbols stay there. 
This same thing happened even if I used <leader>cn.

**Comment by alerque** [comment]
I know this seems unintuitive, but comment syntax is actually a lot more complicated that people realize.

In the case of this example, the best thing to do is not use the minimal comment syntax, use the alternate line prefix syntax. You can toggle to the other mode using `\ca` to run `NERDComAltDelim`. Now commenting a line or block of lines will just prefix them with `//` rather  than trying to wrap them with something. This allows easy toggling and can be applied in multiple layers and pealed off one layer at a time when un-commenting.

The syntax you tried to use cannot be nested. It would be invalid to have `/* /* content */ */` which is what would happen using that syntax to add a layer of comment syntax to your block of code which had a comment in it already. The plugin can't just leave you with broken code so it has two options:

1. not comment out that line at all. This would not reverse properly because when you went to toggle the state all the lines would be commented the same way and uncommenting them would result in a line of real comment text being run as code. Not so hot an idea.
1. Change the nested delimiter to something the language parser doesn't choke on and use a 2 step process to be able to get  back to where you started.

The later is what this plugin does. It can't go straight back to the original comment syntax, but if you were to uncomment that line a second time you'd find it puts the syntax back the way it was.

Again, the easy solution is just not o use the minimal comment syntax for full line comments. Use the syntax which is easier to parse then you won't run into issues like this. Save the minimal syntax for things inside a line which have to be done that way and use the alternate syntax.

**Comment by ckyoog** [comment]
@alerque 

>  It can't go straight back to the original comment syntax, but if you were to uncomment that line a second time you'd find it puts the syntax back the way it was.

If I understand this statement of yours correct. What you were saying is like the case below,

Given a **one line** nested comment in C.
```
/* [> x <] */
```
type `\cu`, we'll get
```
[> x <]
```
type `\cu` again (*the second time*, as you said), we'll get
```
/* x */
```
Looks good.
But my question is why the first `\cu` does not change `/* [> x <] */` to `/* x */` directly, why a second `\cu` is needed.

And I noticed, if given a **multi-line** comment in C, we don't have to type `\cu` twice.
```
/*
 * [> x <]
 */
```
type `\cu` only once, we'll get 
```
/* x */
```

**Comment by alerque** [comment]
@ckyoog That's a good question and I don't know the answer off the top of my head. Could you open this as a separate issue so we can track it? Thanks…
